pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -235,24 +240,16 @@ func (l *podNIC) PlugPhase1() error {
 			log.Log.Reason(err).Error("failed to save interface configuration")
 			return errors.CreateCriticalNetworkError(err)
 		}
-	}
-
-	return nil
-}
 
-func ensureDHCP(bindMechanism BindMechanism, podInterfaceName string) error {
-	dhcpStartedFile := fmt.Sprintf("/var/run/kubevirt-private/dhcp_started-%s", podInterfaceName)
-	_, err := os.Stat(dhcpStartedFile)
-	if os.IsNotExist(err) {
-		if err := bindMechanism.startDHCP(); err != nil {
-			return fmt.Errorf("failed to start DHCP server for interface %s", podInterfaceName)
-		}
-		newFile, err := os.Create(dhcpStartedFile)
-		if err != nil {
-			return fmt.Errorf("failed to create dhcp started file %s: %s", dhcpStartedFile, err)
+		// preparePodNetworkInterface must be called *after* the generate
+		// methods since it mutates the pod interface from which those
+		// generator methods get their info from.
+		if err := bindMechanism.preparePodNetworkInterface(); err != nil {
```

Comment:
`storeCachedDomainIface` should be the last thing done in `phase1` otherwise in case the virt-handler crashes the `prepare` won't be invoked.
(storing it after the `prepare` has its own problems - if the virt-handler crash is during the `prepare`, recalling it may cause unpredicted state of the network).
I'm not sure which one of the bugs is worse but let's not solve one by introducing another one.

Reply:
Right! Forgot about that. Got mislead / confused by your previous comment in https://github.com/kubevirt/kubevirt/pull/5607#discussion_r639494231.

Putting it back the way it was.